app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -158,14 +162,40 @@ public static void startService(Context context)
 		try
 		{
 			activityMonitorIntent = new Intent(context, ActivityMonitorService.class);
-			context.startService(activityMonitorIntent);
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+			{
+				validatePermissionsAndAsStartForegroundService(context, activityMonitorIntent);
+			}
+			else
+			{
+				context.startService(activityMonitorIntent);
+			}
 		}
 		catch (Exception e)
 		{
 			reportException(AppUtils.class.getSimpleName(), e, Thread.currentThread());
 		}
 	}
 
+	/*
+		Checks if application is not permitted to show notifications as this permission is necessary to show foreground notifications and Toasts
+		Starts foreground service.
+		Post Alert message if app is not permitted to do so.
+	 */
+	@TargetApi(26)
+	public static void validatePermissionsAndAsStartForegroundService(Context context, Intent activityMonitorIntent)
+	{
+		if (NotificationManagerCompat.from(context).areNotificationsEnabled())
```

Comment:
To make it clear the error is an exception, structure the code like this:

```
		if (!NotificationManagerCompat.from(context).areNotificationsEnabled())
		{
			Logger.logi("Notifications Disabled", context.getString(R.string.notification_disabled_message));
			displayErrorAlert(context, new ErrorMessage(context.getString(R.string.notification_disabled_message)));
			return;
		}
		context.startForegroundService(activityMonitorIntent);
```

Reply:
Done.